deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/DeployTagWorker.java

```diff
@@ -42,6 +42,19 @@ private void processEachEnvironConstraint(DeployConstraintBean bean) throws Exce
         String tagName = bean.getConstraint_key();
         String envId = environBean.getEnv_id();
         Collection<HostBean> hostBeans = hostDAO.getHostsByEnvId(envId);
+        Collection<String> excludedHostNames = hostDAO.getExcludedHostsFromEnvByEnvId(envId);
```

Comment:
Could you give a bit more description here? Are we just excluding syncing tags? How do we handle the case that host has been added and removed? Do we remove the previous tags?